dn_util.go

```diff
@@ -0,0 +1,216 @@
+package ldap
+
+import (
+	enchex "encoding/hex"
+	"errors"
+	"strings"
+)
+
+// When true, uses strings.EqualFold to compare the values of an RDN.
+// This is usually needed as true for most values, set to false when
+// your DNs contain case sensitive values.
+var RDNCompareFold bool = true
```

Comment:
Is it possible to have this option not be a global variable? An Equals impl depending on external global state seems weird to me.